src/api/server.rkt

```diff
@@ -119,26 +146,99 @@
+    ["julia" (hasheq 'result (core->julia input "expr") 'language to-language)]
+    ["matlab" (hasheq 'result (core->matlab input "expr") 'language to-language)]
+    ["wls" (hasheq 'result (core->wls input "expr") 'language to-language)]
+    ["tex" (hasheq 'result (core->tex input "expr") 'language to-language)]
+    ["js" (hasheq 'result (core->js input "expr") 'language to-language)]
+    [_ (error "Unsupported target language:" to-language)]))
 
 (define (manager-ask msg . args)
   (log "Asking manager: ~a, ~a.\n" msg args)
-  (manager-ask-with-callback msg args))
+  (if manager
+      (manager-ask-with-callback msg args)
+      (match (list* msg args) ; public commands
+        [(list 'wait hash-false job-id)
+         (define result (hash-ref completed-work job-id #f))
+         (match result
+           [#f
+            (match (hash-ref completed-translations job-id #f)
+              [#f (log "Job ~a not complete\n" job-id)]
+              [t
+               (log "Done waiting for translation: ~a\n" job-id)
+               t])]
+           [result
+            (log "Done waiting for job: ~a\n" job-id)
+            result])]
+        [(list 'result job-id) (hash-ref completed-work job-id #f)]
+        [(list 'timeline job-id) (hash-ref completed-work job-id #f)]
+        [(list 'check job-id) (if (hash-ref completed-work job-id #f) job-id #f)]
+        [(list 'count) (list 0 0)]
+        [(list 'improve)
+         (for/list ([(job-id result) (in-hash completed-work)]
+                    #:when (equal? (hash-ref result 'command) "improve"))
+           (get-table-data-from-hash result (make-path job-id)))])))
+
+(define (herbie-do-server-job command job-id)
+  (define herbie-result (wrapper-run-herbie command job-id))
+  (match-define (job-result kind test status time _ _ backend) herbie-result)
+  (define out-result
+    (match kind
+      ['alternatives (make-alternatives-result herbie-result test job-id)]
+      ['evaluate (make-calculate-result herbie-result job-id)]
+      ['cost (make-cost-result herbie-result job-id)]
+      ['errors (make-error-result herbie-result job-id)]
+      ['exacts (make-exacts-result herbie-result job-id)]
+      ['improve (make-improve-result herbie-result test job-id)]
+      ['local-error (make-local-error-result herbie-result job-id)]
+      ['explanations (make-explanation-result herbie-result job-id)]
+      ['sample (make-sample-result herbie-result test job-id)]
+      [_ (error 'compute-result "unknown command ~a" kind)]))
+  out-result)
+
+;; These are split for the 'improve call.
+(define completed-work (make-hash))
+(define completed-translations (make-hash))
 
 (define (manager-ask-with-callback msg args)
   (define-values (a b) (place-channel))
   (place-channel-put manager (list* msg b args))
   (place-channel-get a))
 
 (define (is-server-up)
-  (not (sync/timeout 0 manager-dead-event)))
+  (if manager
+      (not (sync/timeout 0 manager-dead-event))
+      #t))
```

Comment:
Gotta merge with `main`, I think this part of the diff is all from `no-place` being merged.